packages/subcommands/src/lib/utils/SubcommandsHandler.ts

```diff
@@ -0,0 +1,121 @@
+}
+
+export class SubcommandsHandler {
+	// eslint-disable-next-line @typescript-eslint/naming-convention
+	public readonly subcommandsPluginOptions: Required<SubcommandsPluginOptions>;
+
+	public constructor(private readonly client: SapphireClient, options: SubcommandsPluginOptions) {
+		this.subcommandsPluginOptions = {
+			continueParent: options?.continueParent ?? false,
+			runParentAfterSubcommand: options?.runParentAfterSubcommand ?? false,
+			overlappedPreconditions: options?.overlappedPreconditions ?? []
+		};
+	}
+
+	public async resolveSubcommand({ command, message, parameters }: ResolveSubcommandPayload): Promise<ResolveSubcommandResult> {
+		const args = await command.preParse(message, parameters);
+		const store = command.store as Store<Command>;
+		let subcommandFromArgs = await args.pickResult('string');
+		let iterableCommand = command;
+		do {
+			if (!subcommandFromArgs.success || !iterableCommand.subcommands) break;
+
+			const soughtName = subcommandFromArgs.value.toLowerCase();
+
+			const soughtSubcommand = store.find((c) => c.name === soughtName || c.aliases.includes(soughtName));
+			if (!soughtSubcommand || soughtSubcommand.parentName !== iterableCommand.name) break;
+			iterableCommand = soughtSubcommand;
+		} while ((subcommandFromArgs = await args.pickResult('string')));
+
+		return { subcommand: iterableCommand === command ? undefined : iterableCommand, args };
+	}
+
+	public buildSubcommandsTree() {
+		const { commands } = this.client;
+		for (const [name, command] of commands.entries()) {
+			if (!command.subcommands.length) continue;
+
+			for (const subcommandName of command.subcommands) {
+				const subcommand = subcommandName && commands.get(subcommandName);
+				if (!subcommand) throw new SubcommandNotLoadedError(subcommandName ?? '[empty name]', command.name);
+
+				subcommand.parentName = name;
+
+				// Setting up parent's preconditions to found subcommand with filtering
+				let parentPreconditionFiltered: IPreconditionContainer | null = command.preconditions;
+				for (const preconditionName of this.subcommandsPluginOptions.overlappedPreconditions) {
+					parentPreconditionFiltered = this.preconditionsDeepFilter(parentPreconditionFiltered, preconditionName);
+				}
+
+				parentPreconditionFiltered && (subcommand.preconditions as PreconditionContainerArray).add(parentPreconditionFiltered);
+			}
+		}
+	}
+
+	private preconditionsDeepFilter(container: IPreconditionContainer | null, filterTarget: string) {
+		if (container instanceof PreconditionContainerArray) {
+			let containerArray = null;
+			for (const ent of container.entries) {
+				const result = this.preconditionsDeepFilter(ent, filterTarget);
+
+				if (!result) continue;
+				!containerArray && (containerArray = new PreconditionContainerArray());
+
+				containerArray?.add(result);
+			}
+			return containerArray;
+		} else if (container instanceof PreconditionContainerSingle) {
+			return container.name !== filterTarget ? container : null;
```

Comment:
Switching around the condition would fix the [lint warning](https://github.com/sapphire-project/plugins/pull/36/checks?check_run_id=1637234049).